src/redux/effects/effects-for-services/Effect-service.ts

```diff
@@ -0,0 +1,18 @@
+import { Dispatch } from "redux";
+import { getCategoryService } from "../../../services/apiServices/getCategoryService";
+import { getCategoryDropdownAction } from "../../actions/book/bookAction";
+import { iGetCategoryOptionsActionType } from "../../types/book/bookType";
+import { customDispatch } from "../../../services/api-manager/apiUtilities";
+import { iApiResponse } from "../../../services/api-manager/apiManagerInterfaces";
+
+export const getCategoryDropDownEffect =
+  () =>
+  async (
+    dispatch: Dispatch<iGetCategoryOptionsActionType>
+  ): Promise<iApiResponse> => {
+    const response = await getCategoryService();
+    return customDispatch(
```

Comment:
kindly change name of the `customDispatch` it is conflicting with `globalDispatch`.